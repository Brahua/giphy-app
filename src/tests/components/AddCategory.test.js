import React from 'react';
import {shallow} from 'enzyme';
import AddCategory from './../../components/AddCategory'
import '@testing-library/jest-dom'

describe('Tests in <AddCategory>', () => {
  const setCategories = jest.fn();
  const component = <AddCategory setCategories={setCategories}></AddCategory>;
  let wrapper = shallow(component);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(component);
  });

  test('should create the component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  // test('should change search input text ', () => {
  //   const input = wrapper.find('input');
  //   input.simulate('change', { target: { value: ''}});
    
  // });

  test('should NOT post form information after submit', () => {
    wrapper.find('form').simulate('submit', { preventDefault: () => {} });
    expect(setCategories).not.toHaveBeenCalled();
  });
  
  test('should post form information after submit', () => {
    //1. simular el inputChange
    wrapper.find('input').simulate('change', { target: { value: 'Prueba'}});
    //2. simular el submit
    wrapper.find('form').simulate('submit', { preventDefault: () => {} });
    //3. setCategories debe ser llamado
    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    //4. se limpia el input
    expect(wrapper.find('input').prop('value')).toBe('');
  });


})
