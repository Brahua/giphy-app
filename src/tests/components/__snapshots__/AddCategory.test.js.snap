// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <AddCategory> should create the component correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="form-control"
    onChange={[Function]}
    placeholder="Ingrese una palabra para buscar los gifs relacionados"
    type="text"
    value=""
  />
</form>
`;
